tests/Doctrine/Tests/Common/Cache/MemcacheCacheTest.php

```diff
@@ -51,4 +51,34 @@ protected function _getCacheDriver()
         $driver->setMemcache($this->memcache);
         return $driver;
     }
+    
+    /**
+     * The following values get converted to FALSE if you cast them to a boolean.
+     * @see http://php.net/manual/en/types.comparisons.php
+     */
+    public function falseCastedValuesProvider()
+    {
+        return array(
+            array(false),
+            array(null),
+            array(array()),
+            array('0'),
+            array(0),
+            array(0.0),
+            array('')
+        );
+    }
+    
+    /**
+     * Check to see that, even if the user saves a value in memcache that can be interpreted as false,
+     * memcache will still recognize that it's there.
+     * @dataProvider falseCastedValuesProvider
+     */
+    public function testFalseCastedValues($value)
```

Comment:
Please add this test to the common test so that other drivers are also checked against this potential bug (given that all drivers implement the same feature, there should not be many implementation-specific tests given they should all behave the same)